ratis-datastream/src/main/java/org/apache/ratis/datastream/server/DataStreamServer.java

```diff
@@ -0,0 +1,30 @@
+package org.apache.ratis.datastream.server;
+
+import org.apache.ratis.datastream.objects.DataStreamReply;
+import org.apache.ratis.datastream.objects.DataStreamRequest;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A server interface handling incoming streams
+ * Relays those streams to other servers after persisting
+ * It will have an associated Netty client, server for streaming and listening.
+ */
+public interface DataStreamServer {
+
+  /**
+   * Invoked from the server to persist data and add to relay queue.
+   */
+  boolean persistAndAdd(DataStreamRequest request);
```

Comment:
rename to receiveRequest (the implementation could decide what to do)

Also change this and all other methods to return CompletableFuture.